stream_alert_cli/terraform_generate.py

```diff
@@ -0,0 +1,193 @@
+      'output_s3_buckets': '${var.aws-s3}',
+      'input_sns_topics': '${var.aws-sns}'
+    }
+
+    # CloudWatch monitoring module
+    cluster_dict['module']['cloudwatch_monitoring_{}'.format(cluster_name)] = {
+      'source': 'modules/tf_stream_alert_monitoring',
+      'sns_topic_arn': '${{module.stream_alert_{}.sns_topic_arn}}'.format(cluster_name),
+      'lambda_functions': [
+        '{}_{}_streamalert_rule_processor'.format(prefix, cluster_name),
+        '{}_{}_streamalert_alert_processor'.format(prefix, cluster_name)
+      ],
+      'kinesis_stream': '{}_{}_stream_alert_kinesis'.format(prefix, cluster_name)
+    }
+
+    # Add outputs
+    for output in ('username', 'access_key_id', 'secret_key'):        
+        cluster_dict['output']['kinesis_{}_{}'.format(cluster_name, output)] = {
+            'value': '${{module.kinesis_{}.{}}}'.format(cluster_name, output)
+        }
+
+    # Kinesis module
+    cluster_dict['module']['kinesis_{}'.format(cluster_name)] = {
+        'source': 'modules/tf_stream_alert_kinesis',
+        'account_id': '${lookup(var.account, "aws_account_id")}',
+        'region': '${{lookup(var.clusters, "{}")}}'.format(cluster_name),
+        'cluster_name': cluster_name,
+        'firehose_s3_bucket_name': '{}.{}.{}'.format(prefix, cluster_name, firehose_suffix),
+        'stream_name': '{}_{}_stream_alert_kinesis'.format(prefix, cluster_name),
+        'firehose_name': '{}_{}_stream_alert_firehose'.format(prefix, cluster_name),
+        'username': '{}_{}_stream_alert_user'.format(prefix, cluster_name),
+        'stream_config': '${{var.kinesis_streams_config["{}"]}}'.format(cluster_name)
+    }
+
+    # Kinesis events module
+    cluster_dict['module']['kinesis_events_{}'.format(cluster_name)] = {
+        'source': 'modules/tf_stream_alert_kinesis_events',
+        'lambda_production_enabled': True,
+        'lambda_role_id': '${{module.stream_alert_{}.lambda_role_id}}'.format(cluster_name),
+        'lambda_function_arn': '${{module.stream_alert_{}.lambda_arn}}'.format(cluster_name),
+        'kinesis_stream_arn': '${{module.kinesis_{}.arn}}'.format(cluster_name),
+        'role_policy_prefix': cluster_name
+    }
+
+    return cluster_dict
+
+def terraform_generate(**kwargs):
+    """Generate all Terraform plans for the clusters in variables.json"""
+    LOGGER_CLI.info('Generating Terraform files')
+    config = kwargs.get('config')
+    init = kwargs.get('init', False)
+
+    # Setup main
+    main_json = json.dumps(
+        generate_main(init=init, config=config),
+        indent=4,
+        sort_keys=True
+    )
+    with open('terraform/main.tf', 'w') as tf_file:
+        tf_file.write(main_json)
+
+    # Break out early during the init process, clusters aren't needed yet
+    if init:
+        return True
+
+    # Setup clusters
+    for cluster in config['clusters'].keys():
+        if cluster == 'main':
+            raise InvalidClusterName('Rename cluster "main" to something else!')
```

Comment:
It doesn't look like `InvalidClusterName` is defined in this file? It's defined in `runner.py`